src/client/datascience/variablesView/variableView.ts

```diff
@@ -78,6 +78,7 @@ export class VariableView extends WebviewViewHost<IVariableViewPanelMapping> imp
         this.notebookWatcher.onDidExecuteActiveNotebook(this.activeNotebookExecuted, this, this.disposables);
         this.notebookWatcher.onDidChangeActiveNotebook(this.activeNotebookChanged, this, this.disposables);
         this.notebookWatcher.onDidRestartActiveNotebook(this.activeNotebookRestarted, this, this.disposables);
+        this.variables.refreshRequired(this.sendRefreshMessage, this, this.disposables);
```

Comment:
Somethings isn't right here.

Either this was completely broken or we have some weird code.
I'm not sure how this worked in the previous WebView version of Run by line & interactive window.
Did that old code perform their own request in some other place? Or do we have some other pattern or way to make this request (

I.e. i'd like to understand how the old code worked and why we need this here.

Reply:
I believe we need this here because the old code did a refresh when the debugger updated the variables pane. So interactive debug would auto refresh whenever the debugger refreshed. 

Run by line in the webview didn't have this problem.

David and I chatted about something similar to this. The debuggerVariables (or perhaps a standalone class) could be modified to force variable refresh whenver the debugger hits a stop event. I think that would also alleviate this problem.

Reply:
this was removed